frontend/packages/db-structure/src/parser/drizzle/parser.ts

```diff
@@ -0,0 +1,1166 @@
+import * as ts from 'typescript'
+import type {
+  Column,
+  Columns,
+  Constraints,
+  ForeignKeyConstraint,
+  Index,
+  Table,
+} from '../../schema/index.js'
+import type { Processor, ProcessResult } from '../types.js'
+import {
+  convertDefaultValue,
+  convertDrizzleTypeToPgType,
+  convertReferenceOption,
+} from './convertToPgType.js'
+
+interface DrizzleTableDefinition {
+  name: string
+  columns: Record<string, DrizzleColumnDefinition>
+  indexes: Record<string, DrizzleIndexDefinition>
+  compositePrimaryKey?: CompositePrimaryKeyDefinition
+  comment?: string | undefined
+}
+
+interface DrizzleColumnDefinition {
+  name: string
+  type: string
+  typeOptions?: Record<string, unknown>
+  notNull: boolean
+  primaryKey: boolean
+  unique: boolean
+  default?: unknown
+  comment?: string | undefined
+  references?:
+    | {
+        table: string
+        column: string
+        onDelete?: string | undefined
+        onUpdate?: string | undefined
+      }
+    | undefined
+}
+
+interface DrizzleIndexDefinition {
+  name: string
+  columns: string[]
+  unique: boolean
+  type?: string
+}
+
+interface DrizzleEnumDefinition {
+  name: string
+  values: string[]
+}
+
+interface DrizzleRelationDefinition {
+  fromTable: string
+  toTable: string
+  type: 'one' | 'many'
+  fields?: string[]
+  references?: string[]
+}
+
+interface CompositePrimaryKeyDefinition {
+  columns: string[]
+}
+
+interface MethodCall {
+  name: string
+  args: ts.Expression[]
+}
+
+// TypeScript target constants
+const TS_SCRIPT_TARGETS = {
+  FALLBACK: ts.ScriptTarget.ES5,
+  PREFERRED: ts.ScriptTarget.ES2020,
+  LATEST: ts.ScriptTarget.Latest,
+} as const
+
+/**
+ * Get the appropriate TypeScript script target
+ */
+const getScriptTarget = (): ts.ScriptTarget => {
+  if (TS_SCRIPT_TARGETS.PREFERRED !== undefined) {
+    return TS_SCRIPT_TARGETS.PREFERRED
+  }
+  if (TS_SCRIPT_TARGETS.LATEST !== undefined) {
+    return TS_SCRIPT_TARGETS.LATEST
+  }
+  return TS_SCRIPT_TARGETS.FALLBACK
+}
+
+/**
+ * Parse Drizzle TypeScript schema to extract table definitions
+ */
+const parseDrizzleSchema = (
+  sourceCode: string,
+): {
+  tables: Record<string, DrizzleTableDefinition>
+  enums: Record<string, DrizzleEnumDefinition>
+  relations: DrizzleRelationDefinition[]
+} => {
+  try {
+    // Check if TypeScript is available
+    if (!ts || typeof ts.createSourceFile !== 'function') {
+      throw new Error(
+        'TypeScript is not available. Please ensure TypeScript is installed.',
+      )
+    }
```

Comment:
The module resolution of the esm at runtime should give me an error first, so I guess I can turn it off.